docs/contracts/inheritance.rst

```diff
@@ -168,6 +173,58 @@ not known in the context of the class where it is used,
 although its type is known. This is similar for ordinary
 virtual method lookup.
 
+.. _function-overriding:
+
+.. index:: ! overriding;function
+
+Function Overriding
+===================
+
+Base functions can be overridden by inheriting contracts to change their
+behavior. The overriding function must then use the ``override`` keyword in the
+function header as shown in this example:
+
+::
+
+    pragma solidity >=0.5.0 <0.7.0;
+
+    contract Base
+    {
+        function foo() public {}
+    }
+
+    contract Middle is Base {}
+
+    contract Inherited is Middle
+    {
+        function foo() public override {}
+    }
+
+For multiple inheritance, the most derived base contracts that defines the same
+function must be specified explicitly. Additionally, the contract deriving from multiple bases
+defining the same function must explicitly override those functions:
+
+::
+
+    pragma solidity >=0.5.0 <0.7.0;
+
+    contract Base1
+    {
+        function foo() public {}
+    }
+
+    contract Base2
+    {
+        function foo() public {}
+    }
+
+    contract Inherited is Base1, Base2
+    {
+        // Derives from multiple bases defining foo(), so we must explicitly
+        // override it
+        function foo() public override(Base1, Base2) {}
+    }
+
```

Comment:
Can you also add an example (and a test if we do not have it yet) like this:
```
contract A { function f() public pure{} }
contract B is A {}
contract C is A {}
contract D is B, C {}
```

The function that D inherits via B and C is exactly the same function `f` defined in the same base and thus it does not have to be explicitly overridden.